pycqed/analysis_v2/coherence_analysis.py

```diff
@@ -907,150 +1197,404 @@ def prepare_plots(self):
-    arch_model.set_param_hint('dac0', value=0.1, min=0)
+def fit_frequencies(dac, freq,
+                    Ec_guess=260e6, Ej_guess=19e9, offset_guess=0,
+                    dac0_guess=0.5):
+    """
+    Perform fit against the transmon flux arc model.
+
+    Args:
+        dac: flux in units of φ0
+        freq: 01 transition frequency in Hz
 
-    arch_model.make_params()
-    # print('freq, dac', freq, dac)
-    fit_result_arch = arch_model.fit(freq, dac=dac)
+    return:
+        fit_result_arch: an lmfit fit_result object.
+    """
+    # define the model (from the function) used to fit data
+    arch_model = lmfit.Model(arch)
+
+    # set some hardcoded guesses
+    arch_model.set_param_hint('Ec', value=Ec_guess, min=100e6, max=350e6)
+    arch_model.set_param_hint('Ej', value=Ej_guess, min=0.1e9, max=30e9)
+    arch_model.set_param_hint(
+        'offset', value=offset_guess, min=-0.05, max=0.05)
+    arch_model.set_param_hint('dac0', value=dac0_guess, min=0)
+
+    params = arch_model.make_params()
+    fit_result_arch = arch_model.fit(freq, dac=dac, params=params)
     return fit_result_arch
 
 
 def residual_Gamma(pars_dict, sensitivity, Gamma_phi_ramsey, Gamma_phi_echo):
+    """
+    Residual function for fitting dephasing rates (Gamma).
+
+    Two separate linear models are used for the ramsey and echo dephasing
+    rates.
+    """
+    # FIXME: this really needs a dostring to explain what it does
     slope_ramsey = pars_dict['slope_ramsey']
     slope_echo = pars_dict['slope_echo']
     intercept = pars_dict['intercept']
 
-    gamma_values_ramsey = slope_ramsey * np.abs(sensitivity) + intercept
-    # print(len(Gamma_phi_ramsey), len(gamma_values_ramsey))
+    gamma_values_ramsey = slope_ramsey*np.abs(sensitivity) + intercept
     residual_ramsey = Gamma_phi_ramsey - gamma_values_ramsey
 
-    gamma_values_echo = slope_echo * np.abs(sensitivity) + intercept
+    gamma_values_echo = slope_echo*np.abs(sensitivity) + intercept
     residual_echo = Gamma_phi_echo - gamma_values_echo
 
     return np.concatenate((residual_ramsey, residual_echo))
 
 
-def fit_gammas(sensitivity, Gamma_phi_ramsey, Gamma_phi_echo, verbose: bool = False):
-    # create a parametrrer set for the initial guess
+def fit_gammas(sensitivity, Gamma_phi_ramsey, Gamma_phi_echo,
+               verbose: int=0):
+    """
+    Perform a fit to the residual_Gamma using hardcoded guesses.
+
+    Args:
+        sensitivity         x-values
+        Gamma_phi_ramsey    dephasing rate of Ramsey experiment
+        gamma_phi_echo      dephasing rate of echo experiment
+        verbose (int)       verbosity level
+    Returns:
+        fit_result_gammas  an lmfit fit_res object
+    """
+    # create a parameter set for the initial guess
     p = lmfit.Parameters()
     p.add('slope_ramsey', value=100.0, vary=True)
     p.add('slope_echo', value=100.0, vary=True)
     p.add('intercept', value=100.0, vary=True)
```

Comment:
Should there be a lower bound of 0 on the intercept?

Reply:
yes.